database/driver.mjs

```diff
@@ -116,6 +116,57 @@ class dbDriver {
     async find(query) {
         return this.controller.find(query).catch(err => err)
     }
+    async findOne(query) {
+        return this.controller.findOne(query).catch(err => err)
+    }
+
+    /**
+     * Get a database record by its ID.
+     * @param id The ID of the record to retrieve.
+     * @collection collection The collection or table to search.
+     * @return JSON of the matched document or standard error object
+     */
+    async getById(id, collection) {
+        return this.controller.getById(id, collection).catch(err => err)
+    }
+    /**
+     * 
+     * @param {*} payload body of project to be saved
+     * @param {*} collection collection of table
+     * @returns 
+     */
+
+    async saveProject(payload, collection) {
+        return this.controller.saveProject(payload, collection).catch(err => err)
+    }
+
+    /**
+     * Reserve a valid ID from the database for use in building a record 
+     * without collision.
+     * @param seed A seed to base the ID on, depending on the driver.
+     * @return The reserved ID or error JSON
+     */
+    async reserveId(seed) {
+        return this.controller.reserveId(seed).catch(err => err)
+    }
+
+    /**
+     * Check if the submitted chars represent a valid id in the current controller.
+     * @param id The id to check.
+     * @return boolean
+     */
+    isValidId(id) {
+        return this.controller.isValidId(id)
+    }
+
+    /**
+     * Create a valid ID from a supplied string or number.
+     * @param id The id to validify.
+     * @return string | number
+     */
+    asValidId(id) {
```

Comment:
maybe this can be combined with `reserveId(seed)`

Reply:
remove `asValidId()` wherever it appears for "reserveId()"

Reply:
Suspicious that asValidId is actually unused.
![image](https://github.com/user-attachments/assets/fa8ae02a-3fd8-474f-b67b-de1ed84b1262)

Reply:
can `reserveId() still replace asValidId() in the contexts above? @cubap @thehabes